### ISSUE TITLE
Incorporate GFortran problem matcher definition (when written by user) into the upcoming release

### ISSUE THREAD

**Issue by mjaj** [microsoft/vscode-makefile-tools#419]
Like it was reported by a different user in https://github.com/microsoft/vscode-makefile-tools/issues/376, I do not see any errors and warnings (which are reported in the terminal window, and are clickable there) in the problems windows.
My configuration is Windows, MSYS2, using gfortran 12.2.0, Visual Studio Code 1.75.1, Makefile tools 0.6.0.

I also see the error in the output terminal which has been reported by several other users (https://github.com/microsoft/vscode-makefile-tools/issues/392):
```
Error: the task 'makefile: Makefile Tools Build Task' neither specifies a command nor a dependsOn property. The task will be ignored. Its definition is:
{
    "type": "shell",
    "id": "shell,C:\\Programme\\msys64\\ucrt64\\bin\\make.exe,all,-f,C:\\Users\\xxx\\Documents\\Git\\projectname\\_build-gfortran\\subproject\\Makefile,-C,C:\\Users\\jum\\Documents\\Git\\projectname\\_build-gfortran\\subproject,",
    "problemMatcher": [],
    "label": "makefile: Makefile Tools Build Task"
} 
```
I typically use the icon to build.

I don't know whether it is of importance or not, but I note here the line `"problemMatcher": []`.

**Comment by andreeis** [comment]
@mjaj, in order to have errors and warnings show up automatically as a result of our extension analysis, you shouldn't have missing or empty problemMatcher. Define it as below (it's actually the default that will be filled as you start typing it). Please confirm how this works after you specify msvc and/or gcc problem matcher:

          "problemMatchers": [
            "$gcc",
            "$msvc"
          ],

MSVC will be able to parse cl.exe output on windows and GCC will parse gcc (of course) and a few other main ones. There is need for more custom support for other compilers and more problem matchers will be written in future. Hopefully these two will work for you now.

**Comment by andreeis** [comment]
@mjaj, my mistake. You are not defining "problemMatcher" as empty array. You were doing copy paste from the message with the task error which I cannot reproduce (which I also said on the other bug report).
Are you defining "problemMatcher" in your .vscode/settings.json via "makefile.configurations[]" yourself or not? My answer is still correct in case you define "problemMatcher" as empty or you don't define it at all. But if you hit the task error and that error message is mentioning an empty problem matcher array then I still have to investigate, first I need to reproduce it.

**Comment by mjaj** [comment]
Thanks for the quick reply. Here is some more information on my setup:
I start by adding a folder to a new workspace, then set manually (as the Makefile is located in a different folder) "makefile.makeDirectory" and "makefile.makefilePath". Then, I need to run Makefile:Configure from the Command Palette. In the Makefile Project Outline, there is only  Default configuration to choose, and after setting the target, I can start the build process. This results in the message shown above in the output window, and errors don't show up in the problem window.

When I define an own configuration, following the suggestion in your first reply, and select this configuration in the Project Outline, I see the following in the output window:
```
Error: the task 'makefile: Makefile Tools Build Task' neither specifies a command nor a dependsOn property. The task will be ignored. Its definition is:
{
    "type": "shell",
    "id": "shell,C:\\Programme\\msys64\\ucrt64\\bin\\make.exe,all,-f,C:\\Users\\jum\\Documents\\Git\\athlet-utilities\\_build-gfortran\\jacobi\\Makefile,-C,C:\\Users\\jum\\Documents\\Git\\athlet-utilities\\_build-gfortran\\jacobi,",
    "problemMatcher": [
        "$gcc",
        "$msvc"
    ],
    "label": "makefile: Makefile Tools Build Task"
}
```
So, problemMatcher is set, but still no error messages in the problems window.

And here is an example of the error messages as shown in the terminal window (just in case there should be an issue with the format):
```
[ 96%] Building Fortran object jacobi/CMakeFiles/atlj06.dir/src/inp06.F.obj
C:/Users/xxx/Documents/Git/project-name/subproject/src/inp06.F:45:12:

   45 |       app = ''
      |            1
Error: Cannot convert CHARACTER(0) to REAL(4) at (1)
```

Thank you for your effort.

**Comment by mjaj** [comment]
> And here is an example of the error messages as shown in the terminal window (just in case there should be an issue with the format):
> 
> ```
> [ 96%] Building Fortran object subproject/CMakeFiles/atlj06.dir/src/inp06.F.obj
> C:/Users/xxx/Documents/Git/project-name/subproject/src/inp06.F:45:12:
> 
>    45 |       app = ''
>       |            1
> Error: Cannot convert CHARACTER(0) to REAL(4) at (1)
> ```

Meanwhile, I understood that this is output format of gfortran cannot be matched by gcc problemMatcher. So far, however, my attempts to define an own problemMatcher in my own configuration failed (seems to be ignored).

**Comment by andreeis** [comment]
@mjaj, I finally reproduced the build task error. It was printed in the tasks output window. I kept looking for it in the makefile tools output window. From my investigation, there is nothing wrong with how the extension creates the build task and I also found various similar bug reports opened against the vscode team for bogus similar messages that happen for healthy written tasks. I will follow up with the vscode team to confirm. But, even if I do reproduce the message, the task works fine for me, it builds as expected. Same for you? If so, you can ignore the message.

Unfortunately, because some VSCode API limitation around problem matchers, whenever a customer of Makefile Tools wants to define a new problem matcher they have to modify package.json, build a new vsix, try it locally and coordinate with us a new release. Currently there is no other dynamic way for the extension to pick up a problem matcher definition that you would write somewhere else.

You can share with us what you crafted for your problem matcher or try to follow the gcc example in our source code (extension root, package.json).

**Comment by mjaj** [comment]
Yes, l can confirm that this message isn't really connected with an issue in the build process, which works as expected.

Regarding the problem matcher, I didn't make any further attempts. Thanks for the hints how to include a custom  problem matcher. Not sure, but maybe I will give it a try some time.

**Comment by andreeis** [comment]
@mjaj, if VSCode will fix https://github.com/microsoft/vscode/issues/61497, then we can continue with implementing https://github.com/microsoft/vscode-makefile-tools/issues/229 and then you won't need our assistance with incorporating your problem matcher definition into a new release, you will be able to define and use any new problem matcher without a new build of the extension.
